### PR TITLE
Fix Gestures on Popup on ios

### PR DESCRIPTION
**Pull request by inforithmics**
### Description of Change ###

In Create PageHandler a Page is created without setting the Parent, this causes some issues because the Page has no Parent and therefore no Gestures are initialized
Because this is only an issue on iOS I don't know how to create a device specific test. Maybe a Sample is needed.

 ### Linked Issues ###
 

 - Fixes https://github.com/dotnet/maui/issues/15760
 

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs [hunk 1]
```diff
@@ -29,6 +29,11 @@ static PageHandler CreatePageHandler(IPopup virtualView)
 				Content = view
 			};
 
+			if (virtualView.Parent is Element element)
```
Comment (pictos) [inline src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs:32]: don't think we need this `if` check... Every parent should be an `Element`

Comment (inforithmics) [reply src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs]: I only check now for null (Because the Parent is nullable).

Comment (inforithmics) [reply src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs:35]: I added Conditional Compilation for using `AddLogicalChild` as commented here.

https://github.com/dotnet/maui/issues/15760#issuecomment-1676027428

Comment (inforithmics) [reply src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs:35]: Discovered that for .NET 8 a lot of nullable warnings and errors need to be fixed.

for testing i Ignored them by changing following Directory.Build.props (not in pull request).

 `<WarningsAsErrors>nullable,CS0419,CS1570,CS1571,CS1572,CS1573,CS1574,CS1580,CS1581,CS1584,CS1589,CS1590,CS1592,CS1598,CS1658,CS1734</WarningsAsErrors>`

to

 `<WarningsAsErrors>CS0419,CS1570,CS1571,CS1572,CS1573,CS1574,CS1580,CS1581,CS1584,CS1589,CS1590,CS1592,CS1598,CS1658,CS1734</WarningsAsErrors>`

Comment (brminnick) [reply src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs:35]: No worries! We won't be targeting .NET 8 until it GAs in November. Thanks for the heads up!

Comment (pictos) [reply src/CommunityToolkit.Maui/HandlerImplementation/Popup/Popup.macios.cs:33]: @inforithmics, yes it's nullable, so you can set null to it. So you can remove all the if code and just assign the Parent value

**Review by pictos — APPROVED** [review]
@inforithmics thanks for your PR, I took the liberty to change the code since it was simple to do. We don't need to care about .net 8 here, since this code will not land on .NET 8 anyway.

**Comment by inforithmics** [comment]
No problem, this will work in .NET 8 too. 
When migrating to .NET 8 all [element].Parent = [parent] should be replace by [parent].AddLogicalChild([element]) though.

**Pull request merged**